### PR TITLE
added javadoc, removed unused exceptions

### LINKED ISSUE #166: cleanup unused exceptions
in the exception package we have originally defined a lot of exceptions we don't make use of

image

### PR DESCRIPTION
**Pull request by mitch-lbw**
Closes #166 #169

### CODE REVIEW

File: src/main/java/com/kryptokrauts/aeternity/sdk/util/EncodingUtils.java [hunk 1]
```diff
@@ -289,6 +302,13 @@ public static String hashEncode(final byte[] input, final String identifier) {
     return hash;
   }
 
+  /**
+   * generate commitment hash needed for name auctioning
```
Comment (marc0olo) [inline src/main/java/com/kryptokrauts/aeternity/sdk/util/EncodingUtils.java:306]:
```suggestion
   * generate commitment hash needed to pre-claim a name
```

File: src/main/java/com/kryptokrauts/aeternity/sdk/domain/ObjectResultWrapper.java [hunk 2]
```diff
@@ -5,10 +5,8 @@
 import lombok.experimental.SuperBuilder;
 
 /**
- * this object result wrapper introduces the error handling when returning simple string from node
+ * This object result wrapper introduces the error handling when returning simple string from node
```
Comment (marc0olo) [inline src/main/java/com/kryptokrauts/aeternity/sdk/domain/ObjectResultWrapper.java:8]: do we return a string here? is this correct?

File: src/main/java/com/kryptokrauts/aeternity/sdk/service/mdw/MiddlewareService.java [hunk 3]
```diff
@@ -9,12 +9,37 @@
...
 
+  /**
+   * synchronously returns the result object of a name auction
+   *
+   * @return result of {@link NameAuctionResult}
+   */
   NameAuctionResult blockingGetNameAuction(String name);
 
+  /**
+   * synchronously returns the result object of all name auction
```
Comment (marc0olo) [inline src/main/java/com/kryptokrauts/aeternity/sdk/service/mdw/MiddlewareService.java:27]:
```suggestion
   * synchronously returns the result object of all name auctions
```

File: src/main/java/com/kryptokrauts/aeternity/sdk/service/mdw/MiddlewareService.java [hunk 4]
```diff
@@ -9,12 +9,37 @@
...
+   *
+   * @return result of {@link NameAuctionsResult}
+   */
   NameAuctionsResult blockingGetNameAuctions();
 
+  /**
+   * synchronously returns the result object of all name auction filtered and paginated by the given
```
Comment (marc0olo) [inline src/main/java/com/kryptokrauts/aeternity/sdk/service/mdw/MiddlewareService.java:34]:
```suggestion
   * synchronously returns the result object of all name auctions filtered and paginated by the given
```

**Review by marc0olo — COMMENTED** [review]
generally looks good. have some suggestions and one question

File: src/main/java/com/kryptokrauts/aeternity/sdk/domain/ObjectResultWrapper.java (hunk 2 above)
Comment (mitch-lbw) [reply src/main/java/com/kryptokrauts/aeternity/sdk/domain/ObjectResultWrapper.java:8]: copy paste from the StringResultWrapper ;)

**Review by marc0olo — APPROVED** [review]

**Pull request merged**